conversation_reconstruction_dataflow/conversation_reconstruction/dataflow_main.py

```diff
@@ -11,81 +11,100 @@
 import subprocess
 import json
 from os import path
+import urllib2
+import traceback
 
 import apache_beam as beam
 from apache_beam.options.pipeline_options import PipelineOptions
 from apache_beam.options.pipeline_options import SetupOptions
-from apache_beam.io import ReadFromText
-from apache_beam.io import WriteToText
-from google.cloud import storage
-import traceback
-
+from apache_beam.io.gcp import bigquery as bigquery_io 
 
 def run(known_args, pipeline_args):
   """Main entry point; defines and runs the reconstruction pipeline."""
 
+  pipeline_args.extend([
+    '--runner=DataflowRunner',
+    '--project=wikidetox-viz',
+    '--staging_location=gs://wikidetox-viz-dataflow/staging',
+    '--temp_location=gs://wikidetox-viz-dataflow/tmp',
+    '--job_name=yiqing-reconstruction-test',
+    '--num_workers=30',
+  ])
+
+
   pipeline_options = PipelineOptions(pipeline_args)
   with beam.Pipeline(options=pipeline_options) as p:
 
     # Read the text file[pattern] into a PCollection.
-    filenames = (p | ReadFromText(known_args.input)
+    filenames = (p | beam.io.Read(beam.io.BigQuerySource(query='SELECT UNIQUE(page_id) as page_id FROM [%s]'%known_args.input_table, validate=True)) 
                    | beam.ParDo(ReconstructConversation())
-                   | beam.io.Write(bigquery.BigQuerySink(known_args.output_table, schema=known_args.schema, validate=True)))
+                   | beam.io.Write(bigquery_io.BigQuerySink(known_args.output_table, schema=known_args.output_schema, validate=True)))
 
 class ReconstructConversation(beam.DoFn):
-  def process(self, element):
+  def process(self, row):
 
-    from construct_utils import constructing_pipeline
+#    from construct_utils import constructing_pipeline
     import logging
-    import json
+    from google.cloud import bigquery as bigquery_op 
+    input_table = "wikidetox-viz:wikidetox_conversations.test_page_3_issue21" 
```

Comment:
I was trying to make this as a parameter, but failed to pass into the method, any suggestion how to do this?